### ISSUE TITLE
The action should run eslint for all files if `.eslintrc` is among of the changed files

### ISSUE THREAD

**Issue by owanturist** [sibiraj-s/action-eslint#11]
Hi there,

Thank you for creating such a nice github action! It works pretty smoothly except when the `.eslintrc` file appears among the changed ones. In the worst-case scenario, it does not run eslint at all (when only the rc file is changed, like enabling a new rule), in the best case it runs only on changed files and might miss some potential eslint rules violations.

Do you think it would make sense to run eslint for all (respecting the listed extensions in the github action) files in case `.eslintrc` is changed? It might complicate the flow a bit by the fact that `.eslintrc` might have a variety of extensions like `.eslintrc.json`, `.eslintrc.js` or just `.eslintrc`

**Comment by sibiraj-s** [comment]
Hi.

Thank you and thanks for creating this issue. I am not sure why that happens. I'll have to check. Do you have any reproducible repo?

Also, I have some thoughts let me debug some more on whats happening before answering to this.

**Comment by sibiraj-s** [comment]
Just read that `.eslintrc` config file is deprecated a long back, at-least 4 years back. May be eslint doesn't do a great job with that? 🤔 Or do you have issues with eslintrc without extension as well?

**Comment by sibiraj-s** [comment]
May be I can add an option for config file

```yaml
      - uses: sibiraj-s/action-eslint@v2
        with:
          github-token: ${{ secrets.GITHUB_TOKEN }} # Optional
          eslint-args: '--ignore-path=.gitignore --quiet'
          extensions: 'js,jsx,ts,tsx'
          annotations: true
          config-file: .eslintrc.json
```

Will this fix the issue?

**Comment by sibiraj-s** [comment]
I misunderstood the issue earlier. Running the entire eslint is a easy thing, but things might complicate when its included in this plugin as that was never its intention

The following might help?

```yaml
jobs:
  lint:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v3
    - uses: dorny/paths-filter@v2
      id: filter
      with:
        filters: |
          eslintrc:
            - '.eslintrc*'

    # run eslint on entire directory if eslintrc is changed
    - name: Run eslint
      if: steps.filter.outputs.eslintrc == 'true'
      run: npx eslint .

    # run eslint on changed files
    - name: Run eslint on changed files
      if: steps.filter.outputs.eslintrc == 'false'
      uses: sibiraj-s/action-eslint@v2
```

**Comment by owanturist** [comment]
Hi @sibiraj-s 

Thanks for the quick response!

> Running the entire eslint is a easy thing, but things might complicate when its included in this plugin as that was never its intention

Yeah, that's the point - eslint does not care where it's been running, so it is not aware if the `.eslintrc*` file changed or not compared to the latest run. I ended up introducing a similar workaround [you proposed](https://github.com/sibiraj-s/action-eslint/issues/11#issuecomment-1637356272) - check if the `.eslintrc*` file was changed and run eslint either manually for all files or via a github action.

In the end, I used another [github action](https://github.com/marketplace/actions/eslint-changed-files) but it also does not support the case I have reported. But what it does have is `all_files` flag that can be used for running eslint for all files in case `.eslintr*` has changed. It makes the job config a bit easier and more consistent (one step vs two).

---

On the other hand, this challenge is relatively easy to solve. There are a few options:

1. Detect if the changed files have `.eslintrc*`  file among them. If found, run the `eslint` command with `files = []` [here](https://github.com/sibiraj-s/action-eslint/blob/c8d96d780af17293d59608d02ead57763f2eb9b3/src/eslint.ts#L35)
2. Introduce `config_path` field, so it is easier to detect the rc file change because it would be clear where to look at.  If found, run the `eslint` command with `files = []` [here](https://github.com/sibiraj-s/action-eslint/blob/c8d96d780af17293d59608d02ead57763f2eb9b3/src/eslint.ts#L35)
3. Introduce `all_files` so the workaround you proposed is easier to implement. 

It is possible to apply all of the listed options, which makes the action's API more flexible and intelligent: whenever `all_files` is `true` - run eslint for all files, OR if the `config_path` is provided, detect the provided path changes and run eslint for all files if so, OR  detect changes of any `.eslintrc*` files and run eslint for all files.

---

Nevertheless, the workaround solves the issue and I hope my feedback is valuable and can improve UX of the action.
